src/main/scala/mesosphere/marathon/state/RootGroup.scala

```diff
@@ -285,18 +290,35 @@ class RootGroup(
         case (_, group) => group.id != newPod.id || group.containsApps || group.containsPods
       },
       dependencies = oldGroup.dependencies,
-      version = version)
+      version = version,
+      enforceRole = oldGroup.enforceRole)
     putGroup(newGroup, version)
   }
 
+  /**
+    * Update a group with the specified group id by applying the update function.
+    *
+    * It has the same semantics as [[updateApp()]] and [[updatePod()]]. If the group does not exist
+    * it will be created. The update does *not* change the group version.
+    *
+    * @param groupId the if od the group to be updated.
+    * @param fn the update function.
+    * @return the new root group with the update group.
+    */
+  def updateGroup(groupId: PathId, fn: Option[Group] => Group): RootGroup = {
+    val updatedGroup = fn(group(groupId))
```

Comment:
In updateApp and updatePod we have a 
```
require(updatedGroup.id == oldGroup.id, "Group Id must not be changed by fn")
```
Might be a good idea to add that here as well